### DISCUSSION TITLE
Role is not deleted :|

### DISCUSSION THREAD

**Discussion by troller2017** [spatie/laravel-permission#2629] [Q&A]
Hello I am testing something.. is their other methods to do this? the role is not being removed when I do this..
```
$per = Permission::where('name', 'some_permission')->firstOrFail();
$per->delete(); // <-- this works

Role::create(['name' => 'some_role'); // <-- this works

$role = Role::where('name', 'myrole')->firstOrFail();
$result = $role->delete(); // <-- this does nothing

```

**Comment by drbyte** [comment]
In this package, the Permission and Role models are Laravel Eloquent models. Operations like `->delete()` are handled by Eloquent.

So yes, calling `$role->delete()` where `$role` is actually a valid Eloquent object, should work.
So ... maybe your `$role` isn't actually an Eloquent object?
